src/pybind/mgr/dashboard/tox.ini

```diff
@@ -177,3 +178,13 @@ description = Generate Sphinx documentation from OpenAPI specification
 deps = -r../../../../admin/doc-requirements.txt
 changedir = ../../../../doc
 commands = sphinx-build -W -b html -c . -D suppress_warnings=ref.* -d {envtmpdir}/doctrees mgr/ceph_api {envtmpdir}/html
+
+[testenv:gen-nvmeof-proto]
+allowlist_externals = cp
+changedir = ../
+description = Generate Python code from Protobuf files for nvmeof gateway
+sitepackages = true
+commands_pre = cp ../../nvmeof/gateway/control/proto/gateway.proto dashboard/services/proto/
+commands =
+    python -c "import grpc; print(f'using grpc version: {grpc.__version__}')"
+    python -m grpc_tools.protoc --proto_path=. --python_out=. --grpc_python_out=. dashboard/services/proto/gateway.proto --experimental_allow_proto3_optional
```

Comment:
`experimental_allow_proto3_optional` surprisingly is from 2020's, so is it still an experimental flag??

Reply:
somehow it gave me an error and told me to use it. Let me try again.

Reply:
looks like it depends on the grpc version installed in the centos. Probably can be removed in the newer ones I guess but since the proto file is using the proto3 optional or experimental features (like `optional`) we'll have to generate it with this flag for now.